homeassistant/components/varta_storage/sensor.py

```diff
@@ -0,0 +1,74 @@
+"""Sensor platform of the VARTA Storage integration."""
+
+from homeassistant.components.sensor import SensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Initialize the integration."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        VartaStorageEntity(coordinator, idx) for idx, ent in enumerate(coordinator.data)
+    )
+
+
+class VartaStorageEntity(CoordinatorEntity, SensorEntity):
+    """An entity using CoordinatorEntity.
+
+    The CoordinatorEntity class provides:
+      should_poll
+      async_update
+      async_added_to_hass
+      available
+
+    """
+
+    def __init__(self, coordinator, idx):
+        """Pass coordinator to CoordinatorEntity."""
+        super().__init__(coordinator)
+        self.idx = idx
+
+        self._attr_device_info = DeviceInfo(
+            configuration_url="http://" + coordinator.config_entry.data["host"],
+            identifiers={(DOMAIN, str(coordinator.config_entry.unique_id))},
+            manufacturer="VARTA",
+            name="VARTA Battery",
+        )
+
+        self._attr_unique_id = coordinator.config_entry.unique_id + "-" + self.name
+
+    @property
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        return self.coordinator.data[self.idx]["name"]
+
+    @property
+    # pylint: disable=overridden-final-method
+    def state(self):
+        """Return the state of the sensor."""
+        return self.coordinator.data[self.idx]["state"]
```

Comment:
You should not define the `state` attribute yourself: it is automatically computed by HA. I guess what you are putting in `state` here should rather go in the `native_value` attribute which should contain the sensors current measurement in the unit given by the attribute `native_unit_of_measurement`.

Reply:
got it. This also explains why i struggled with the pylint errors and needed to disable it for this part of the code in the previous version of the sensor.

Changed it according to your recommendation, thanks!